src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeLoader.cs

```diff
@@ -34,5 +43,48 @@ public ResourceTypeComponents LoadType(ResourceTypeReference reference)
 
             return resourceTypeFactory.GetResourceType(serializedResourceType);
         }
+
+        public ResourceTypeComponents? LoadFallbackResourceType()
+        {
+            if (fallbackResourceType != null)
+            {
+                var serializedResourceType = typeLoader.LoadResourceType(fallbackResourceType);
+
+                return resourceTypeFactory.GetResourceType(serializedResourceType);
+            }
+
+            // No fallback type provided in JSON
+            return null;
+        }
+
+        public NamespaceConfiguration? LoadNamespaceConfiguration()
+        {
+            if (typeSettings != null)
+            {
+                var name = typeSettings.Name;
+                var version = typeSettings.Version;
+                var isSingleton = typeSettings.IsSingleton;
+
+                TypeSymbol? configurationType = null;
+
+                if (typeSettings.ConfigurationType != null)
+                {
+                    var reference = typeSettings.ConfigurationType;
```

Comment:
The following is semantically equivalent: 

```suggestion
                if (typeSettings.ConfigurationType is {}  reference)
                {
```